rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+  const SmallSearchInput = loadable(() => import('@razorpay/blade/components'));
+
+  const SearchScreen = () => (isMobile ? <SmallSearchInput /> : <LargeSearchInput />);
+
+  export default SearchScreen;
+  ```
+
+- Consider, another example where we want to show and hide bunch of components based on different screen sizes.
+  <img alt="Responsive Modal" src="./images/responsive-adaptive-layout/adaptive-layout-with-multiple-layouts.png">
+
+  <br/>
+
+  Here's how we'll implement the above mockup in code
+
+  ```jsx
+  import loadable from '@loadable/component';
+  import { Stack, Card, Title, Hidden, Avatar, Divider } from '@razorpay/blade/components';
+
+  const DesktopNavBar = loadable(() => import('@razorpay/blade/components'));
+  const MobileNavBar = loadable(() => import('@razorpay/blade/components'));
+  const BottomNavBar = loadable(() => import('@razorpay/blade/components'));
+
+  const Dashboard = () => (
+    <Stack direction={{ s: 'column', l: 'row' }}>
+      {isMobile ? <MobileNavBar /> : <DesktopNavbar />}
+      <Stack direction="column">
+        <Hidden hide={{ s: true, l: false }}>
+          <Stack direction="row" justifyContent="space-between">
+            <div>some profile information</div>
+            <Stack direction="row">
+              <Avatar />
+              <Avatar />
+            </Stack>
+          </Stack>
+        </Hidden>
+        <Stack direction={{ s: 'column', l: 'row' }}>
+          <Card>
+            <Title>This is Card Title</Title>
+            <Content>This is Card Content</Content>
+          </Card>
+          <Card>
+            <Title>This is Card Title</Title>
+            <Content>This is Card Content</Content>
+          </Card>
+          <Card>
+            <Title>This is Card Title</Title>
+            <Content>This is Card Content</Content>
+          </Card>
+        </Stack>
+      </Stack>
+      <Hidden hide={{ s: false, l: true }}>
+        <BottomNavBar />
+      </Hidden>
+    </Stack>
+  );
+  ```
+
+  So here we have introduced a new utility component called `Hidden` which accepts responsive props and enables us to hide/show component based on different screen sizes
+
+### On Blade Design Side
+
+While building components on Figma we have to keep in mind following things:
+
+1. If the component is responsive, then design it for all the screens and properly define/document the responsive properties.
+
+   - `width`
+   - `height`
+   - `font-size`
+   - `padding`
+   - `margin`
+
+2. If the component is adaptive, then design it for all the screens and explicitly mention about the multiple variations of a component for different screen sizes.
+
+### On Consumer Design Side
+
+When designing layouts and consuming components from Design System Library on Figma make sure to do following things:
+
+1. Design the mockups for all the different screen sizes.
+2. Pick the right version of the component for respective screen sizes.
```

Comment:
By using internal component `variants`.

Reply:
didn't understand this part? 🤔

Reply:
I meant to say instead of writing:
> Pick the right version of the component for respective screen sizes.

We can end the statement by saying:
> Pick the right version of the component for respective screen sizes using component variants in Figma.